**Ques 1. Explain the properties of the F-distribution.**

Ans 1)
Properties of F-Distribution:

1. Non-Symmetry: The F-distribution is skewed to the right.

2. Continuous: The F-distribution is a continuous distribution.

3. Positive: The F-distribution takes only positive values.

4. Two Parameters: The F-distribution has two parameters: degrees of freedom for the numerator (df1) and degrees of freedom for the denominator (df2).

5. Shape: The shape of the F-distribution changes with df1 and df2.



***
**Ques 2.In which types of statistical tests is the F-distribution used, and why is it appropriate for these tests?**

Ans 2)
1. Testing for Equality of Variances:

F-test for Equal Variances:

 This test compares the variances of two populations. The null hypothesis is that the two populations have equal variances. The F-statistic is calculated as the ratio of the sample variances. If the calculated F-statistic is significantly larger than the critical value, we reject the null hypothesis and conclude that the variances are not equal.


2. Analysis of Variance (ANOVA):

One-way ANOVA: This test compares the means of multiple groups. The null hypothesis is that all group means are equal. The F-statistic is calculated as the ratio of the mean square between groups (MSB) to the mean square within groups (MSW). If the calculated F-statistic is significantly larger than the critical value, we reject the null hypothesis and conclude that at least one group mean is different from the others.

***
**Ques 3. What are the key assumptions required for conducting an F-test to compare the variances of two
populations?**

Ans 3)
To conduct an F-test to compare the variances of two populations, the following key assumptions must be met:

Independence: The two samples must be independent of each other. This means that the selection of one sample should not influence the selection of the other.

Normality: Both populations from which the samples are drawn should be normally distributed. This assumption is crucial, as the F-test is sensitive to departures from normality.   

Equal Variances (Homoscedasticity): This assumption, although counterintuitive for a test of variances, is necessary for the validity of the F-test.1 Essentially, we assume that the two populations have equal variances.   


***
**Ques 4. What is the purpose of ANOVA, and how does it differ from a t-test?**

Ans 4)
Analysis of Variance (ANOVA) is a statistical technique used to compare the means of two or more groups. It helps determine whether there are significant differences among the means of these groups. 

Difference between ANOVA and t-test are:

t-test: Used to compare the means of two groups.   

ANOVA: Used to compare the means of more than two groups.

***
**Ques 5. Explain when and why you would use a one-way ANOVA instead of multiple t-tests when comparing more
than two groups.**

Ans 5)
When comparing the means of more than two groups, it's generally more appropriate to use a one-way ANOVA rather than multiple t-tests. Here's why:   

1. Controlling Type I Error Rate:

Multiple Comparisons Problem: Conducting multiple t-tests increases the likelihood of making a Type I error (false positive). With each test, there's a chance of incorrectly rejecting the null hypothesis.   
ANOVA's Advantage: ANOVA controls the overall Type I error rate, ensuring that the probability of making at least one Type I error across all comparisons is at the specified significance level (usually 0.05).   

2. Increased Statistical Power:

Pooling Variance: ANOVA pools the variance from all groups to estimate a common population variance, leading to a more precise estimate and increased statistical power.   
Efficiency: This pooled variance allows for a more sensitive test, making it easier to detect significant differences between group means.

3. Comprehensive Analysis:

Overall Significance: ANOVA provides an overall test of whether there are any differences among the group means.   
Pairwise Comparisons: If the ANOVA is significant, post-hoc tests like Tukey's HSD or Bonferroni correction can be used to identify specific pairwise differences between groups, controlling for the Type I error rate.

 

***
**Ques 6. Explain how variance is partitioned in ANOVA into between-group variance and within-group variance.
How does this partitioning contribute to the calculation of the F-statistic?**

Ans 6)

Between-Group Variance: This measures the variability between the means of different groups. It represents the differences in the average values of the dependent variable across the different groups. 
  

Within-Group Variance: This measures the variability within each group. It represents the natural variation or noise within each group, which is not attributable to the group differences.

The F-statistic is calculated by comparing the between-group variance to the within-group variance. Specifically, it is the ratio of the mean square between groups (MSB) to the mean square within groups (MSW):   

F = MSB / MSW

where,
MSB: This represents the average variability between the group means.

   
MSW: This represents the average variability within each group.   


A larger F-statistic indicates that the between-group variance is significantly larger than the within-group variance. This suggests that the differences between the group means are not due to chance but are likely real effects of the independent variable.

***
**Ques 7. Compare the classical (frequentist) approach to ANOVA with the Bayesian approach. What are the key
differences in terms of how they handle uncertainty, parameter estimation, and hypothesis testing?**

Ans 7)
Classical (Frequentist) vs. Bayesian ANOVA:

**Uncertainty:**

Classical: Treats parameters as fixed, unknown quantities. Uncertainty is expressed in terms of confidence intervals, which represent the range of values within which the true parameter is likely to fall with a certain level of confidence.   

Bayesian: Treats parameters as random variables. Uncertainty is quantified using probability distributions, specifically the posterior distribution, which represents the updated beliefs about the parameter given the observed data and prior beliefs.   

**Parameter Estimation:**

Classical: Estimates parameters using point estimates (e.g., sample mean, sample variance) and constructs confidence intervals around these point estimates.

Bayesian: Estimates parameters by calculating the posterior distribution, which represents the full range of plausible values for the parameter, along with their associated probabilities.

**Hypothesis Testing:**

Classical: Formulates null and alternative hypotheses and calculates a p-value, which represents the probability of observing the data (or more extreme data) if the null hypothesis were true. A small p-value leads to rejection of the null hypothesis.   

Bayesian: Directly calculates the probability of the null hypothesis being true, given the observed data and prior beliefs. This is often expressed as a Bayes factor, which compares the evidence for the null hypothesis to the evidence for the alternative hypothesis.   


***
Ques 8. Question: You have two sets of data representing the incomes of two different professions1

Profession A: [48, 52, 55, 60, 62]

Profession B: [45, 50, 55, 52, 47]

**Perform an F-test to determine if the variances of the two professions'
incomes are equal. What are your conclusions based on the F-test?**

**Task: Use Python to calculate the F-statistic and p-value for the given data.**

**Objective: Gain experience in performing F-tests and interpreting the results in terms of variance comparison.**

Ans 8)

Interpretation:

The F-statistic (3.23) measures the ratio of the variances between the two professions.

The p-value (0.109) is greater than the common significance level of 0.05.

Conclusion: There is no statistically significant difference in the variances of incomes between the two professions. This suggests that the variability in incomes is similar for Profession A and Profession B. ​


In [1]:
from scipy.stats import f_oneway

# Data for the two professions
profession_A = [48, 52, 55, 60, 62]
profession_B = [45, 50, 55, 52, 47]

# Perform the F-test
f_statistic, p_value = f_oneway(profession_A, profession_B)

print("F-statistic:", f_statistic)
print("p-value:", p_value)

F-statistic: 3.232989690721649
p-value: 0.10987970118946545


***
Ques 9. Question: Conduct a one-way ANOVA to test whether there are any statistically significant differences in
average heights between three different regions with the following data.

Region A: [160, 162, 165, 158, 164]

Region B: [172, 175, 170, 168, 174]

Region C: [180, 182, 179, 185, 183]

Task: Write Python code to perform the one-way ANOVA and interpret the results

Objective: Learn how to perform one-way ANOVA using Python and interpret F-statistic and p-value

In [ ]:
from scipy.stats import f_oneway

# Data for the three regions
region_A = [160, 162, 165, 158, 164]
region_B = [172, 175, 170, 168, 174]
region_C = [180, 182, 179, 185, 183]

# Perform one-way ANOVA
f_statistic, p_value = f_oneway(region_A, region_B, region_C)

print("F-statistic:", f_statistic)
print("p-value:", p_value)

F-statistic: 67.87330316742101
p-value: 2.8706641879370266e-07


Ans 9)
The F-statistic (67.87) indicates how much the group means differ relative to the variability within the groups.

The p-value (2.87×10^-7) is significantly less than the common significance level of 0.05.

Conclusion: There is a statistically significant difference in average heights between the three regions. This suggests that at least one region's average height is different from the others